linalg/sparsemat.cpp

```diff
@@ -660,65 +668,70 @@ void SparseMatrix::AddMult(const Vector &x, Vector &y, const double a) const
 
    // Skip if matrix has no non-zeros
    if (nnz == 0) {return;}
-   if (Device::Allows(Backend::CUDA_MASK) && useCuSparse)
+   if ((Device::Allows(Backend::CUDA_MASK) || Device::Allows(Backend::HIP_MASK)) &&
+       useGPUSparse)
    {
-#ifdef MFEM_USE_CUDA
+#ifdef CUDA_or_HIP
       const double alpha = a;
       const double beta  = 1.0;
 
       // Setup descriptors
       if (!initBuffers)
       {
-#if CUDA_VERSION >= 10010
+#if CUDA_VERSION >= 10010 || defined(MFEM_USE_HIP)
          // Setup matrix descriptor
-         cusparseCreateCsr(&matA_descr,Height(), Width(), J.Capacity(),
-                           const_cast<int *>(d_I),
-                           const_cast<int *>(d_J), const_cast<double *>(d_A), CUSPARSE_INDEX_32I,
-                           CUSPARSE_INDEX_32I, CUSPARSE_INDEX_BASE_ZERO, CUDA_R_64F);
+         cu_or_hip(sparseCreateCsr)(&matA_descr,Height(), Width(), J.Capacity(),
+                                    const_cast<int *>(d_I),
+                                    const_cast<int *>(d_J), const_cast<double *>(d_A), CU_or_HIP(SPARSE_INDEX_32I),
+                                    CU_or_HIP(SPARSE_INDEX_32I), CU_or_HIP(SPARSE_INDEX_BASE_ZERO),
+                                    CUDA_or_HIP(_R_64F));
```

Comment:
One argument per line will be more readable I think.

Reply:
I agree

Reply:
done